src/librustc_trans/cabi_x86.rs

```diff
@@ -100,17 +101,19 @@ pub fn compute_abi_info<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>,
         let mut free_regs = 2;
 
         for arg in &mut fty.args {
-            if arg.is_ignore() || arg.is_indirect() { continue; }
+            let attrs = match arg.mode {
+                PassMode::Direct(ref mut attrs) => attrs,
```

Comment:
PassMode::Cast can't occur here because? Is this before we do casts? Could you add an assertion here to prevent unniceness?

Reply:
I could add a `bug!("x86 never casts arguments")` (if you look at the rest of the file, only return values are cast).